src/main/kotlin/no/nav/syfo/melding/MeldingService.kt

```diff
@@ -21,10 +25,17 @@ class MeldingService(
         // TODO: legg til i joark-cronjob og send på kafka her
     }
 
-    fun getMeldingerTilBehandler(personIdent: PersonIdent): List<MeldingTilBehandler> {
-        // TODO: Her ville vi vel gruppere det på en eller annen måte for ulike samtaler/dialoger
-        return database.getMeldingerForArbeidstaker(personIdent)
-            .filter { !it.innkommende } // Foreløpig bare ta MeldingTilBehandler-meldingene
-            .map { it.toMeldingTilBehandler() }
+    fun getConversationMeldingerMap(personIdent: PersonIdent): Map<UUID, List<Melding>> {
+        val meldinger = database.getMeldingerForArbeidstaker(personIdent)
+        return meldinger.groupBy(
+            keySelector = { it.conversationRef },
+            valueTransform = {
+                if (it.innkommende) {
+                    it.toMeldingFraBehandler().toMelding(it.behandlerRef!!)
+                } else {
+                    it.toMeldingTilBehandler().toMelding()
+                }
+            }
+        )
```

Comment:
Her er vi i domenelaget, og går direkte til å returnere `Melding` som egt er tilknyttet DTOen. Men vi kan ikke ha `List<MeldingTilBehandler | MeldingFraBehandler>` i kotlin, så det vet ikke helt om man skal returene noe annet enn dette fra servicen uten å gjøre det ekstra komplisert.

Reply:
Kan vi lage en dto som er generisk og kan brukes både for MeldingTilBehandler og MeldingFraBehandler?